### PR TITLE
Initial pull request for openmp_reduction and dpc_reduce examples

### PR DESCRIPTION
**Pull request by terdner**
# Description

Initial pull request for openMP_reductino and dpc_reduce examples.  

Fixes # (issue) 

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Sample Migration (Moving sample from old repository after completing checklist established)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Command Line
- [x] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode

# Checklist for Moving samples:
Links and Details can be found in the samples WG Teams Files. 

- [x] Review sample design with domain reviewers https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts 
- [x] Implement coding guidelines and ensure code quality.
- [x] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [x] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
- [x] Adhere to readme template 
- [x] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [x] Enforce format via clang-format config file
- [x] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [x] Review readme with Tom Lenth or Joe Oster. (GitHub User: JoeOster)
- [ ] Tested using Dev Cloud when applicable
- [ ] Implement fixes for ONSAM Jiras
- [ ] If you have new dependencies/binaries, inform Samples Project Manager Swapna R Dontharaju (@srdontha) or @JoeOster

### CODE REVIEW

File: DirectProgramming/DPC++/ParallelPatterns/dpc_reduce/sample.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ 	"guid": "ECF6C8EB-753B-4107-AF64-60662CE41726",
+ 	"name": "DPC Reduce",
+ 	"categories": ["Toolkit/Intel® oneAPI Base Toolkit/oneAPI DPC++ Compiler/oneAPI DPC++ Library/CPU and GPU"],
+ 	"description": "It models transform reduce in different ways showing capability of oneAPI.",
+ 	"toolchain": ["dpcpp"],
+ 	"languages": [{
+ 		"cpp": {}
+ 	}],
+ 	"targetDevice": ["CPU", "GPU"],
+ 	"os": ["linux"],
```
Comment (praveenkk123) [inline DirectProgramming/DPC++/ParallelPatterns/dpc_reduce/sample.json:11]: Missing "ciTests" field in the sample.json.  "ciTests": Please look at simple-add example for reference

File: DirectProgramming/DPC++/ParallelPatterns/openmp_reduction/sample.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
+  "guid": "3C2AA983-1B31-4C05-BCA0-724F159B8535", 
+  "name": "openmp_reduction",
+  "categories": ["Toolkit/Intel® oneAPI Base Toolkit/oneAPI DPC++ Compiler/oneAPI DPC++ Library/CPU and GPU"],
+  "description": "It models transform reduce in different ways showing capability of oneAPI.",
+  "toolchain": ["icpx"],
+  "languages": [{"cpp":{}}],
+  "targetDevice": ["CPU", "GPU"],
+  "os": ["linux"],
```
Comment (praveenkk123) [inline DirectProgramming/DPC++/ParallelPatterns/openmp_reduction/sample.json:9]: Missing "ciTests" field in the sample.json.  "ciTests": Please look at simple-add example for reference

**Comment by JoeOster** [comment]
> ## Test execution status
> See detailed test results for specific OS below.

@mkitez - citests are in place, but Linux didn't run. I also don't see the report in TeamCity

**Comment by anjgola** [comment]
@JoeOster  Can you please review as well. @praveenkk123 If this meets all the checklist please click on 'merge and squash' and go ahead and merge it.

File: DirectProgramming/DPC++/ParallelPatterns/openmp_reduction/License.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Copyright 2020 Intel Corporation
```
Comment (JoeOster) [inline DirectProgramming/DPC++/ParallelPatterns/openmp_reduction/License.txt:1]: Please remove the date, to adhere to the new standards

File: DirectProgramming/DPC++/ParallelPatterns/dpc_reduce/License.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Copyright 2020 Intel Corporation
```
Comment (JoeOster) [inline DirectProgramming/DPC++/ParallelPatterns/dpc_reduce/License.txt:1]: Please remove date to adhere to the new standards

**Comment by terdner** [comment]
@JoeOster - removed dates from the License.txt file and pushed to master.

**Comment by praveenkk123** [comment]
@pmpeter1 , Hi Paul can you please confirm if you have reviewed and you are OK with these samples

**Comment by JoeOster** [comment]
> Here are a few things that need to be fixed
> 
> * chrono commands should go through dpc_common
> * The use of '\n' is recommended over std::endl except if the explicit flush is desired
> * Needs a License.txt file in the root of both
> 
> samples

These issues were resolved

**Comment by terdner** [comment]
> The dpc_common includes need the comments about how to find this shared file.
> 
> Would be good to rename the functions in the DPC++ sample to also include the "purpose" of the functions in the name and not just the execution mode like is done in the OpenMP sample.

@pmpeter1  - these two items are now fixed.

**Pull request merged**